### DISCUSSION TITLE
What is the best way to use this without committing `flake.nix` to the repository?

### DISCUSSION THREAD

**Discussion by sylvesterroos** [shanesveller/nix-beam-flakes#28] [Q&A] [answered]
I can't simply add files like these to the repository, as it's a company project. If I add `flake.nix` to `.git/info/exclude` I get an error saying that the git tree is dirty.

Thanks :)

**Comment by sylvesterroos** [comment, marked as answer]
You can't make this stuff up. Right after making this post, I come across [this](https://discourse.nixos.org/t/can-i-use-flakes-within-a-git-repo-without-committing-flake-nix/18196/48) forum post.

Solution: `use flake path:.`

**Comment by shanesveller** [comment]
In case it's helpful to you or anyone searching:

Historically in this scenario I've used a flake managed as a separate repo entirely, and then called `nix develop <absolute-or-relative-path>`. The approach works well with direnv as well, which can likewise be written as `use flake <relative-or-absolute-path>` and then you can just gitignore `.envrc`. Optionally you can use the `.git/info/exclude` file to keep even that much out of the primary repository.

This strategy can also be used for multiple disparate projects that have very uniform requirements, all pointing to the same out-of-tree flake. Sadly I almost never get that much homogeneity in my own work.

**Reply by sylvesterroos** [reply]
Great! In my case a local flake works better since that way I can read from the local `.tool-versions`, but it's good to know there are other ways!
